bowtie/_core.py

```diff
@@ -625,6 +643,144 @@ async def start(
 
         yield cls(harness=harness, id=id, info=info, reporter=reporter)
 
+    @classmethod
+    async def download_versions_of(cls, id: ConnectableId) -> Set[str] | None:
+        url = HOMEPAGE / "implementations" / id / "matrix-versions.json"
+
+        async with httpx.AsyncClient(timeout=10) as client:
+            try:
+                with Progress(
+                    TextColumn("[bold blue]{task.description}"),
+                    SpinnerColumn(finished_text=""),
+                    BarColumn(bar_width=None),
+                    TextColumn(
+                        "[progress.percentage]{task.percentage:>3.0f}%",
+                    ),
+                    "•",
+                    DownloadColumn(),
+                    "•",
+                    TimeElapsedColumn(),
+                    console=Console(),
+                    transient=True,
+                ) as progress:
+                    task = progress.add_task(
+                        description=f"Fetching versions of {id}",
+                        total=None,
+                    )
+
+                    response = await client.get(str(url))
+                    response.raise_for_status()
+
+                    content = response.content
+                    content_length = len(content)
+
+                    progress.update(
+                        task,
+                        description=(
+                            f"Successfully fetched all versions of {id}!"
+                        ),
+                        completed=content_length,
+                        total=content_length,
+                        advance=content_length,
+                    )
+                    await asyncio.sleep(2)
+
+                    return frozenset(json.loads(content))
+            except httpx.HTTPStatusError:
+                return None
+
+    @classmethod
+    async def download_reports_for(
```

Comment:
This seems like a weird thing to be a classmethod -- it doesn't construct an instance of the class, and doesn't really seem to use `Implementation` (`cls`) at all.